draft-ietf-suit-trust-domains.md

```diff
@@ -214,72 +216,37 @@ If a Recipient supports groups of interdependent components (a Component Set), t
-
-~~~
-current := components\[component-index\]
-    if component-index is not false
-    else dependencies\[dependency-index\]
-~~~
-
-As a result, Set Component Index is described as current := components\[arg\]. The actual operation performed for Set Component Index is described by the following pseudocode, however, because of the definition of current (above), these are semantically equivalent.
-
-~~~
-component-index := arg
-dependency-index := false
-~~~
-
-Similarly, Set Dependency Index is semantically equivalent to current := dependencies\[arg\], but the actual operation performed is:
+* Two new commands are introduced.
 
-~~~
-dependency-index := arg
-component-index := false
-~~~
-
-Dependencies are identified by digest, but referenced in commands by Dependency Index, the index into the array of Dependencies.
-
-##  Changes to Special Cases of Component Index and Dependency Index {#index-true}
+    * Process dependency.
+    * Is Dependency.
 
-The considerations that apply in Special Cases of Component Index and Dependency Index (Section 6.5) of {{I-D.ietf-suit-manifest}} are augmented to include Dependency Index as well as Component Index.
-
-The target(s) assigned for each command are defined by the following pseudocode.
-
-~~~
-if component-index is true:
-    current-list = components
-else if component-index is array:
-    current-list = [ components[idx] for idx in component-index ]
-else if component-index is integer:
-    current-list = [ components[component-index] ]
-else if dependency-index is true:
-    current-list = dependencies
-else if dependency-index is array:
-    current-list = [ dependencies[idx] for idx in dependency-index ]
-else:
-    current-list = [ dependencies[dependency-index] ]
-for current in current-list:
-    cmd(current)
-~~~
+* Dependency manifests are also components. All commands may target dependency manifests as well as components, with one exception: process dependency. Commands defined outside of this draft and {{I-D.ietf-suit-manifest}} MAY have additional restrictions.
+* Dependencies are processed in lock-step with the Root Manifest. This means that every dependency's current command sequence must be executed before a dependent's later command sequence may be executed. For example, every dependency's Dependency Resolution step MUST be executed before any dependent's payload fetch step.
+* When performing a suit-condition-image-match operation on a component, the manifest processor MUST first determine whether or not the component is a dependency manifest. If identified as a dependency manifest envelope, the manifest processor MUST compute the digest over only the SUIT_Manifest bstr, not the complete SUIT_Manifest_Envelope. This is so that severable elements, added or removed signatures, and delegations do not affect the integrity measurements of the manifest.
```

Comment:
I think it is difficult to determine whether or not the component is a dependency manifest, because the component would be arbitrary binary data. It may be a SUIT_Envelope_Tagged manifest starting from h'D86B', or may not (just a SUIT_Envelope), and not a SUIT_Envelope_Tagged manifest but starting from h'D86B'.
How about adding a parameter like `suit-parameter-is-suit-manifest`?

Reply:
This was the reason for adding the suit-common element that defines which components are dependency manifests. The suit-dependencies element is a map of component index to extra metadata:
`SUIT_Dependencies = { uint => SUIT_Dependency_Metadata }`
`SUIT_Dependency_Metadata = {? suit-dependency-prefix => SUIT_Component_Identifier, $$SUIT_Dependency_Extensions}`

Existence in the `SUIT_Dependencies` map shows that a component is a manifest.
Does that resolve the concern?